### ISSUE TITLE
Calculation of Sikhorn_SEQ

### ISSUE THREAD

**Issue by agokrani** [2018cx/Multi-Level-OT#5]
Hi! 

Thank you for sharing this work. I am currently working on implementing this on my side. I realized that in the sikhorn loss, you are running a loop for computing emd_loss. However, looking at the loop

```
for i in range(p_s.shape[0]):
            emd_loss = 0.001*self.sinkhorn_loss(x=p_s[i],y=p_t[i])
return emd_loss
```

This will only save the loss for last index and not the entire sequence, is this a bug or is there a reason behind this? Besides, I also found that in normalize function the z_scores are computed without subtracting means

```
def normalize(value):
    means = value.mean(dim=-1, keepdim=True)
    stds = value.std(dim=-1, keepdim=True)
    z_score_normalized_student = (value)/ (stds+0.0001)
    return z_score_normalized_student
```

Can you please explain this as well?

**Comment by 2018cx** [comment]
Hi! Thanks a lot for your feedback and for carefully reviewing the code!

You're absolutely right about the emd_loss loop — the uploaded version mistakenly missed the + operator, which caused the loop to only retain the loss from the last index. This has been fixed in the latest commit.

As for the normalize function, the input values will go through a softmax afterward, so subtracting the mean beforehand has no effect on the final result. That’s why we omit it — only scaling by the standard deviation is needed in this case. Thanks again for pointing this out!

**Comment by agokrani** [comment]
@2018cx thank you so much for responding fast. Few other quick questions:

1. Why did you use loop instead of vectorized implementation? 
2. Have you tried doing topk first and then normalization? In my case the seq_len are too large so we need to do topk first before sorting. 
3. Have you tried different weights for each loss?
